packages/nodes-base/nodes/FunctionItem/FunctionItem.node.ts

```diff
@@ -75,22 +75,70 @@ return item;`,
 		};
 
 		for (let itemIndex = 0; itemIndex < length; itemIndex++) {
+			const mode = this.getMode();
+
 			try {
 				item = items[itemIndex];
+				item.index = itemIndex;
 
 				// Copy the items as they may get changed in the functions
 				item = JSON.parse(JSON.stringify(item));
 
 				// Define the global objects for the custom function
 				const sandbox = {
+					/** @deprecated for removal */
```

Comment:
Mention what method to use instead. I know you're sending it to the UI at runtime, but add it in the comment as well, so that it can easily be shown in an IDE.